src/pages/data/index.js

```diff
@@ -34,19 +44,110 @@ const StateList = ({ states, stateData }) => {
   )
 }
 
-const StatesNav = ({ stateList }) => (
-  <Box width={[1, 1, 1, 1 / 2]} className="state-nav" m="0 auto 1rem">
-    <h3>Jump to a state:</h3>
-    <ul>
-      {stateList.map(state => (
-        <li key={state.node.state}>
-          <Link to={`/data#state-${state.node.state}`}>{state.node.state}</Link>
-        </li>
-      ))}
-    </ul>
-  </Box>
+const StatesNoScriptNav = ({ stateList }) => (
+  <>
+    <noscript>
+      <style>{`
+    .state-nav {
+      display: block !important;
+    }
+    `}</style>
+    </noscript>
+    <Box width={[1, 1, 1, 1 / 2]} className="state-nav" m="0 auto 1rem">
+      <h3>Jump to a state:</h3>
+      <ul>
+        {stateList.map(({ node }) => (
+          <li key={node.state}>
+            <a href={`#state-${node.state.toLowerCase()}`}>{node.state}</a>
+          </li>
+        ))}
+      </ul>
+    </Box>
+  </>
 )
 
+const StatesNav = ({ stateList }) => {
+  const [searchTerm, setSearchTerm] = useState('')
+
+  const searchStates = term => {
+    if (term.trim() === '') {
+      return null
+    }
+    const results = []
+    stateList.forEach(({ node }) => {
+      if (node.name.toLowerCase().search(term.toLowerCase()) > -1) {
+        results.push(node)
+      }
+    })
+    return results
+  }
+
+  const results = searchStates(searchTerm)
+
+  return (
+    <div className="state-combobox-nav">
+      <noscript>
+        <style>{`
+    .state-combobox-nav {
+      display: none !important;
+    }
+    `}</style>
+      </noscript>
+      <label htmlFor="jump-to-state">
+        Type a state&apos;s name to jump to it
+      </label>
+      <Combobox
+        openOnFocus
+        onSelect={selectedItem => {
+          const stateId = stateList.find(({ node }) => {
+            return node.name === selectedItem
+          })
+          if (stateId && typeof window !== 'undefined') {
+            window.location.hash = `state-${stateId.node.state.toLowerCase()}`
+          }
+        }}
+      >
+        <ComboboxInput
+          id="jump-to-state"
+          placeholder="State or territory"
```

Comment:
:+1: for not using "state" :smile:

Reply:
Having lived in a territory, I try not to be "state normative."